inference-engine/src/vpu/common/src/ngraph/operations/static_shape_reshape.cpp

```diff
@@ -40,8 +41,11 @@ void StaticShapeReshape::validate_and_infer_types() {
 
     const auto& inputShape = get_input_shape(0);
 
-    auto outputDimensionsValues = ::vpu::evaluateTargetShape(targetShape);
-    NODE_VALIDATION_CHECK(this, !outputDimensionsValues.empty(), "StaticShapeReshape (", get_friendly_name(), ") can't evaluate output shape");
+    ngraph::PartialShape targetPShape;
+    NODE_VALIDATION_CHECK(this, ngraph::evaluate_as_partial_shape(targetShape, targetPShape) && targetPShape.is_static(),
```

Comment:
This is not right to use evaluate_as_partial_shape here. Please see the opset1::Reshape operation validate_and_infer_types. 

Second input of Reshape may contain -1 as value but it is treated as Dimension::dynamic() inside the PartialShape. 

Why can't we trigger opset1::Reshape::validate_and_infer_types and copy only maximum part of each dimension to the output shape of static operation? The same question for TopK and Broadcast.

Reply:
That's a very good point! Fixed.